src/components/Table/sortRows.ts

```diff
@@ -40,8 +40,10 @@ function sortBatch<R extends Kinded>(
       // If both rows are pinned, we don't sort within them, because by pinning the page is taking
       // explicit ownership over the order of the rows (and we also don't support "levels of pins",
       // i.e. for change events putting "just added" rows `pin: last` and the "add new" row `pin: lastest`).
-      const ap = a.pin === "first" ? -1 : a.pin === "last" ? 1 : 0;
-      const bp = b.pin === "first" ? -1 : b.pin === "last" ? 1 : 0;
+      const ap =
```

Comment:
Suggestion: Maybe add a helper method like `getPin(a)` so this could be either:

```
const ap = getPin(a.pin) === "first" ? -1 : getPin(a.pin) === "last" ? 1 : 0;
```

Or:

```
const aAt = getAt(a.pin);
const ap = aPin === "first"  ? -1 : aPin === "last" ? 1 : 0;
const bAt = getAt(b.pin);
```